sky/backends/cloud_vm_ray_backend.py

```diff
@@ -3460,15 +3462,23 @@ def _execute(
         Returns:
             Job id if the task is submitted to the cluster, None otherwise.
         """
-        if task.run is None:
+        if task.run is None and self._setup_cmd is None:
             logger.info('Run commands not specified or empty.')
             return None
-        # Check the task resources vs the cluster resources. Since `sky exec`
-        # will not run the provision and _check_existing_cluster
-        # We need to check ports here since sky.exec shouldn't change resources
-        valid_resource = self.check_resources_fit_cluster(handle,
-                                                          task,
-                                                          check_ports=True)
+        if task.run is None:
+            # If the task has no run command, we still need to execute the
+            # generated ray driver program to run the setup command in detached
+            # mode.
+            # In this case, we don't need to check the resources.
+            valid_resource = sky.Resources()
```

Comment:
If the task has gpu resources requirement, and the setup actually takes gpu to compute (e.g. some simple data preprocessing), should we keep the original resources requirement?

Reply:
The setup script is not using the original task's resources anyway (we don't assign GPUs to setup in the generated program). That said, it will only increase the schedule time for the setup if we keep the task resources for setup only ones.

Reply:
Added some comments : )